nemo/collections/llm/tools/autotuner/README.md

```diff
@@ -0,0 +1,421 @@
+```
+
+### Remote Execution
+
+All operations are executed remotely on Lepton clusters:
+
+1. **Code Packaging**: Local scripts are packaged and uploaded
+2. **Job Creation**: Lepton jobs are created with appropriate resources
+3. **Log Streaming**: Real-time logs are streamed to your terminal
+4. **Result Retrieval**: Results are available in the mounted workspace
+
+### One-time Setup
+
+The launcher automatically handles one-time setup tasks:
+
+- **NeMo Installation**: Local NeMo changes are installed once per workspace
+- **Environment Setup**: Python paths and cache directories are configured
+- **Dependency Management**: All required packages are available in the base image
+
+
+## 🐛 Troubleshooting
+
+### Common Issues
+
+1. **Job Fails Immediately**
+   - Check Lepton authentication: `lepton auth status`
+   - Verify resource availability: `lepton resource list`
+   - Check workspace mounting: Ensure `/nemo-workspace` is accessible
+
+2. **Configuration Generation Fails**
+   - Verify model name is supported: `python launcher.py list-models`
+   - Check resource requirements match available resources
+   - Ensure config directory is writable
+
+3. **Training Experiments Fail**
+   - Check GPU availability and compatibility
+   - Verify data paths and permissions
+   - Review logs for specific error messages
+
+4. **Results Analysis Issues**
+   - Ensure log files exist and are readable
+   - Check log prefix matches actual log files
+   - Verify cost parameters are reasonable
+
+### Getting Help
+
+```bash
+# Show general help
+python launcher.py --help
+
+# Show help for specific command
+python launcher.py generate --help
+python launcher.py run --help
+python launcher.py results --help
+```
+
+### Log Retrieval
+
+If a job fails, you can retrieve logs manually:
+
+```bash
+# Get job logs
+lepton job logs JOB_ID
+
+# List recent jobs
+lepton job list
+
+# Get job details
+lepton job get JOB_ID
+
+### Step 2: Run Training Experiments
+```bash
+# Edit run_experiments.py with correct paths
+python3 run_experiments.py
+```
+
+
+## Configuration Tips for Scripts
+
+### Model Selection
+- Use exact NeMo model names: `"llama2"`, `"mixtral_8x7b"`, `"gemma2_2b"`, etc.
+- Check supported models with: 
+    ```
+    from nemo.collections.llm.tools.autotuner import list_models; list_models()
+    ```
+
+### Resource Configuration
+- `resource_shape`: Use format `"gpu.countxtype"` (e.g., `"gpu.8xh200"`, `"gpu.4xa100"`)
+- Ensure `nodes * gpus_per_node` matches your total GPU count
+- Parallelism sizes must divide evenly into total GPU count
+
+### Memory Considerations
+- AutoTuner automatically detects potential OOM configurations
+- Configurations flagged as OOM risks will be skipped during training
+- Use `run_all=True` to force run all configurations (not recommended)
+
+### Performance Optimization
+- Start with smaller parameter ranges for quick testing
+- Increase `max_steps_per_run` for more accurate performance measurement
```

Comment:
probably add a note that this will increase the amount of time it takes to run though.